src/reformat/rgb_impl.rs

```diff
@@ -450,3 +450,223 @@ pub fn yuv_to_rgb_any(
     }
     Ok(())
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn yuv_to_rgb() {
+        fn create_420(
+            matrix_coefficients: MatrixCoefficients,
+            y: &[&[u8]],
+            u: &[&[u8]],
+            v: &[&[u8]],
+        ) -> image::Image {
+            let mut yuv = image::Image {
+                width: y[0].len() as u32,
+                height: y.len() as u32,
+                depth: 8,
+                yuv_format: PixelFormat::Yuv420,
+                matrix_coefficients,
+                ..Default::default()
+            };
+            assert!(yuv.allocate_planes(decoder::Category::Color).is_ok());
+            for plane in [Plane::Y, Plane::U, Plane::V] {
+                let samples = if plane == Plane::Y {
+                    &y
+                } else if plane == Plane::U {
+                    &u
+                } else {
+                    &v
+                };
+                assert_eq!(yuv.height(plane), samples.len());
+                for y in 0..yuv.height(plane) {
+                    assert_eq!(yuv.width(plane), samples[y].len());
+                    for x in 0..yuv.width(plane) {
+                        yuv.row_mut(plane, y as u32).unwrap()[x] = samples[y][x];
+                    }
+                }
+            }
+            yuv
+        }
+        fn assert_near(yuv: &image::Image, r: &[&[u8]], g: &[&[u8]], b: &[&[u8]]) {
+            let mut dst = rgb::Image::create_from_yuv(&yuv);
+            dst.format = rgb::Format::Rgb;
+            dst.chroma_upsampling = ChromaUpsampling::Bilinear;
+            assert!(dst.allocate().is_ok());
+            assert!(yuv_to_rgb_any(&yuv, &mut dst, AlphaMultiplyMode::NoOp).is_ok());
+            assert_eq!(dst.height, r.len() as u32);
+            assert_eq!(dst.height, g.len() as u32);
+            assert_eq!(dst.height, b.len() as u32);
+            for y in 0..dst.height {
+                assert_eq!(dst.width, r[y as usize].len() as u32);
+                assert_eq!(dst.width, g[y as usize].len() as u32);
+                assert_eq!(dst.width, b[y as usize].len() as u32);
+                for x in 0..dst.width {
+                    let i = (x * dst.pixel_size() + 0) as usize;
+                    let pixel = &dst.row(y).unwrap()[i..i + 3];
+                    assert_eq!(pixel[0], r[y as usize][x as usize]);
+                    assert_eq!(pixel[1], g[y as usize][x as usize]);
+                    assert_eq!(pixel[2], b[y as usize][x as usize]);
+                }
+            }
+        }
+
+        // Testing identity 4:2:0 -> RGB would be simpler to check upsampling
+        // but this is not allowed (not a real use case).
+        assert_near(
+            &create_420(
+                MatrixCoefficients::Bt601,
+                /*y=*/
+                &[
+                    &[0, 100, 200],  //
```

Comment:
are the trailing `//` necessary?

it seems like this may just fit in one line since we use a column limit of 100 for rust source.

Reply:
I wanted to keep the same 2D presentation for each test case.